server/storage/schema/schema.go

```diff
@@ -28,27 +28,39 @@ var (
 	V3_6 = semver.Version{Major: 3, Minor: 6}
 )
 
-// UpdateStorageSchema updates storage version.
-func UpdateStorageSchema(lg *zap.Logger, tx backend.BatchTx) error {
-	tx.Lock()
-	defer tx.Unlock()
-	v, err := detectStorageVersion(lg, tx)
+// Migrate updates storage version to provided target version.
+func Migrate(lg *zap.Logger, tx backend.BatchTx, target semver.Version) error {
```

Comment:
nit: maybe worth to put it in separate migrate.go as it's going to become huge on its own...

Reply:
Ack, when defining schema, will split it from migration.